test/model/autocomplete_test.dart

```diff
@@ -349,4 +320,131 @@ void main() {
       doCheck('Four F', eg.user(fullName: 'Full Name Four Words'), false);
     });
   });
+
+  group('MentionAutocompleteView sorting users results', () {
+    late PerAccountStore store;
+
+    Future<void> prepare({
+      List<User> users = const [],
+      List<RecentDmConversation> dmConversations = const [],
+    }) async {
+      store = eg.store(initialSnapshot: eg.initialSnapshot(
+        recentPrivateConversations: dmConversations));
+      await store.addUsers(users);
+    }
+
+    group('MentionAutocompleteView.compareByDms', () {
+      test('has DMs with userA and userB, latest with userA, prioritizes userA', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [1],    maxMessageId: 200),
+          RecentDmConversation(userIds: [1, 2], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).isNegative();
```

Comment:
These test cases have a fair number of lines that are repetitive. Let's pull those out into a local helper function, like this:
```suggestion
        check(compare1vs2()).isNegative();
```

That'll mean each test case adds a grand total of 7 or 8 lines instead of 11 or 12, which makes the whole list of test cases quite a lot more compact to look at. That's helpful for scanning through it and seeing what's covered and thinking of things that might not be.

Relatedly, it focuses each test case much more tightly on just the information that's distinctive in it, which is helpful for understanding what each individual test case is saying.